kviewmodel/src/commonMain/kotlin/com/adeo/kviewmodel/KViewModel.kt

```diff
@@ -1,17 +1,29 @@
 package com.adeo.kviewmodel
 
+import kotlinx.coroutines.CoroutineExceptionHandler
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.cancel
+import kotlin.coroutines.CoroutineContext
+import kotlin.native.concurrent.ThreadLocal
 
 public abstract class KViewModel {
 
+    private val mainCoroutineExceptionHandler: CoroutineExceptionHandler? by lazy {
+        getCoroutineExceptionHandler() ?: sharedExceptionHandler
+    }
     private val coroutineTags = hashMapOf<String, CoroutineScope>()
+    private val mainCoroutineContext = (SupervisorJob() + Dispatchers.Main.immediate).run {
+        val exceptionHandler = mainCoroutineExceptionHandler ?: return@run this
+        this + exceptionHandler
+    }
```

Comment:
It also confuses me that the `mainCoroutineExceptionHandler` can be changed during life of a viewModel but the `mainCoroutineContext` will remain attached to the original handler.
Although I'm not sure if that would be a problem. Maybe you have some thoughts about this?

Reply:
I think that we can do it for this task. `mainCoroutineExceptionHandler` is `Lazy`, so we will get only one `ExceptionHandler`, and it will not be changed during ViewModel lifecycle, if I'm not mistaken